source/Model.js

```diff
@@ -450,7 +450,8 @@ function testUsername(username, domain) {
 
 // Make sure that the caller id is sent to the API alongside the sharing token
 // by replacing the client separator with the one expected by the API
-function formatSharingCallerId(sharing) {
+function formatSharingURI(sharingURI) {
```

Comment:
Hi Thiébaud, 

What about the string 'foo%2Bbar' - when you uri decode this first, you'll get 'foo+bar', which will be transformed into 'foo bar', right? Even though the '+' was encoded correctly? 

IMHO, the uri decoding should come after '+' removal - the '+' is essentially part of the encoding, not part of the cleartext. 

Am I making this any clearer? 
kaspar